components/elm/src/utils/restUtilMod.F90.in

```diff
@@ -99,36 +99,25 @@ contains
        status = PIO_put_att(ncid, varid, 'interpinic_flag_meanings', &
             "1=nearest neighbor, 2=copy directly, 3=skip")
 
-       if (present(comment)) then
-          call ncd_putatt(ncid, varid, 'comment', trim(comment))
-       end if
-       if (present(units)) then
-          call ncd_putatt(ncid, varid, 'units', trim(units))
-       end if
-       if (present(fill_value)) then
-          call ncd_putatt(ncid, varid, '_FillValue', fill_value, lxtype)
-       else if (lxtype == ncd_double) then
-          call ncd_putatt(ncid, varid, '_FillValue', spval, lxtype)
+       if (.not. present(fill_value)) then
+          if (lxtype == ncd_double) then
+             call ncd_putatt(ncid, varid, '_FillValue', spval, lxtype)
+          end if
        end if
-       if (present(missing_value)) then
-          call ncd_putatt(ncid, varid, 'missing_value', missing_value, lxtype)
-       else if (lxtype == ncd_double) then
-          call ncd_putatt(ncid, varid, 'missing_value', spval, lxtype)
+       if (.not. present(missing_value)) then
+          if (lxtype == ncd_double) then
+             call ncd_putatt(ncid, varid, 'missing_value', spval, lxtype)
+          end if
```

Comment:
What happens in the case `lxtype` is not `ncd_double`?

Reply:
@bishtgautam : It looks like the ncd_defvar() call now handles the case where the user specifies a missing_value (and the old code only supported special fillvalues for double and int missing_values).

Reply:
@jayeshkrishna Actually the new code is still the same as the old code here:
If missing_value is present, it will be used to call ncd_putatt (for the new code this is moved to ncd_defvar).
If missing_value is not present, the special value spval (defined in module elm_varcon) is used by default only when lxtype is ncd_double.

Reply:
> What happens in the case `lxtype` is not `ncd_double`?

@bishtgautam In the case lxtype is not ncd_double, the attribute "missing_value" will not be set for the variable when the optional missing_value argument is NOT present. This behavior is exactly the same as the old one (the special value spval is only used for double type).